[feat] Debug Log Page

samuelkrissi opened the pull request:
## Describe your changes

The Debug Logs Viewer adds a feature to view, filter, analyze, and manage Salesforce debug logs from the extension, with pagination, user/date filters, and intelligent action extraction (Apex classes, triggers, Flows, etc.). It includes grep-like filtering templates, real-time search with highlighting, and Agentforce-powered analysis with customizable instructions for performance, errors, and governor limits. Additional features include bulk delete, one-click download/share, syntax highlighting for logs, and automatic handling of large files (>1.5MB) to prevent browser crashes.

## Checklist before requesting a review
- [x] I have read and understand the [Contributions section](https://github.com/tprouvot/Salesforce-Inspector-reloaded#contributions)
- [x] Target branch is releaseCandidate and not master
- [x] I have performed a self-review of my code
- [x] I ran the [unit tests](https://github.com/tprouvot/Salesforce-Inspector-reloaded#unit-tests) and my PR does not break any tests
- [x] I documented the changes I've made on the [CHANGES.md](https://github.com/tprouvot/Salesforce-Inspector-reloaded/blob/master/CHANGES.md) and followed actual conventions
- [ ] I added a new section on [how-to.md](https://github.com/tprouvot/Salesforce-Inspector-reloaded/blob/master/docs/how-to.md) (optional)

samuelkrissi committed:
debug log page

samuelkrissi committed:
fix pagination and user filter

samuelkrissi committed:
fix log action display trigger

samuelkrissi committed:
title change

samuelkrissi committed:
fix title

samuelkrissi committed:
fix title

samuelkrissi committed:
changes.md

tprouvot commented on addon/debug-log.js:
```diff
@@ -0,0 +1,1151 @@
+    return h("div", {},
+      h(PageHeader, {
+        pageTitle: "Logs Viewer",
+        orgName: this.model.userInfoModel.userInfo,
```
Could you just set model.orgName like it is done for the other pages ?

tprouvot commented on addon/debug-log.js:
```diff
@@ -0,0 +1,1151 @@
+
+// Add CSS for preview search highlights
+const style3 = document.createElement("style");
+style3.textContent = `mark.sfir-highlight{background:#ffe58a;padding:0 .0rem;border-radius:2px} mark.sfir-highlight.current{background:#f8d24e;outline:1px solid #e1b600}`;
```
Could you use css class instead of style property ?

tprouvot commented on CHANGES.md:
```diff
@@ -23,6 +22,7 @@
 - `Show All Data` Support keyboard shortcut to save edited record values [feature 951](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/951) request by [Mohak Gaur](https://github.com/mohakgaurrr)
 - `Documentation` Revised instructions for creating an External Client App, reflecting the deprecation of Connected Apps and added detailed steps for OAuth configuration and known issues related to Incognito mode [bug 962](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/962) (contribution by [Thomas Malidin Delabriere](https://github.com/tmalidin33))
 - `Authentication` Implement OAuth 2.0 Web Server Flow with [Proof Key for Code Exchange (PKCE)](https://help.salesforce.com/s/articleView?id=xcloud.remoteaccess_pkce.htm&type=5) for orgs with API Access Control enabled - [feature 873](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/873) (contribution by [Mehdi Cherfaoui](https://github.com/mehdicherf))
+- `Logs Viewer` preview all logs, share in mail, possibility to see the action call without open it, filter by user/datetime, download and delete selected logs contribution by [Samuel Krissi](https://github.com/samuelkrissi)
```
Could you put this new feature at the top of the file, just under 2.0 version ?

tprouvot commented on addon/debug-log.js:
```diff
@@ -0,0 +1,1151 @@
+      )
+    ),
+    // log body
+    h("pre", {style: {maxHeight: "60vh", overflow: "auto", background: "#f4f6f9", padding: "8px", borderRadius: "4px"}, dangerouslySetInnerHTML: {__html: html}})
```
I added syntax coloration from prismJs for logs

tprouvot commented on addon/debug-log.js:
```diff
@@ -0,0 +1,1151 @@
+    this.spinnerCount++;
+    try {
+      await sfConn.rest(`/services/data/v${apiVersion}/sobjects/ApexLog/${id}`, {method: "DELETE"});
+      await this.fetchLogs(true);
```
When deleting log, could you use the Toast.js component to display the confirmation or the error ?

tprouvot commented on addon/debug-log.js:
```diff
@@ -0,0 +1,1151 @@
+    if (!log) return;
+    (async () => {
+      try {
+        const xhr = await sfConn.rest(`/services/data/v${apiVersion}/tooling/sobjects/ApexLog/${id}/Body`, {responseType: "blob"}, true);
```
I'm unsure about the share feature, especially those days when most of us are relying on collaboration tools like slack or teams. And for those who wants to share it, they can do it manually with the downloaded file.
What do you think about it ?

samuelkrissi replied:
because in compagny like mine in bank because a lot of limitation of security to added it to jira or send mail in very critical bug in production
it's very useful @tprouvot

tprouvot commented on addon/debug-log.js:
```diff
@@ -0,0 +1,1151 @@
+                    log.Status || "-"
+                  )
+                ),
+                h("td", {style: {width: cw.size}}, (log.LogLength/1024|0)),
```
Could you format the log length with two decimal ? It could be useful for size less than 1MB

tprouvot commented on addon/debug-log.js:
```diff
@@ -0,0 +1,1151 @@
+    onMouseDown: (e) => model.onResizeStart(col, e.clientX)
+  });
+  const onMouseMove = (e) => model.onResizeMove(e.clientX);
+  const onMouseUp = () => model.onResizeEnd();
```
I have errors in the console, could you check please ?

`debug-log.js:821 Uncaught TypeError: model.onResizeEnd is not a function
    at onMouseUp (debug-log.js:821:33)`

tprouvot commented on addon/debug-log.js:
```diff
@@ -0,0 +1,1151 @@
+    className: "sfir-col-resizer",
+    onMouseDown: (e) => model.onResizeStart(col, e.clientX)
+  });
+  const onMouseMove = (e) => model.onResizeMove(e.clientX);
```
I have errors in the console, could you check please ?
`debug-log.js:820 Uncaught TypeError: model.onResizeMove is not a function
    at onMouseMove (debug-log.js:820:36)`

tprouvot committed:
Implement code highlighting for debug logs using Prism.js

tprouvot reviewed (changes requested):
Well, this is great @samuelkrissi 👏 🎉 
I love the UI and the code perfectly matches with the extension.

I have added syntax coloration but the search highlight is not working anymore (I'll check that) for log view and did some lint updates.

Could you explain what are those arrows for ? 
<img width="1309" height="277" alt="Image" src="https://github.com/user-attachments/assets/3cab1cce-603a-4c26-9f35-40083636b62e" />

Another thing regarding the search, I think we could add a feature to filter (kind of grep) the log lines to only display the search like t[his vs code plugin](https://marketplace.visualstudio.com/items?itemName=earshinov.filter-lines). We would provide some search templates like 'USER_DEBUG' AND 'EXCEPTION_THROWN' 

Again, great work Samuel !

samuelkrissi commented:
> Well, this is great @samuelkrissi 👏 🎉 I love the UI and the code perfectly matches with the extension.
> 
> I have added syntax coloration but the search highlight is not working anymore (I'll check that) for log view and did some lint updates.
> 
> Could you explain what are those arrows for ? <img alt="Image" width="1309" height="277" src="https://private-user-images.githubusercontent.com/35368290/528688060-3cab1cce-603a-4c26-9f35-40083636b62e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjYyNDU4MDYsIm5iZiI6MTc2NjI0NTUwNiwicGF0aCI6Ii8zNTM2ODI5MC81Mjg2ODgwNjAtM2NhYjFjY2UtNjAzYS00YzI2LTlmMzUtNDAwODM2MzZiNjJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjIwVDE1NDUwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ1NzdhZTVkNDIyZDgwMGRjZDMzNWIyZjkxNTRiMmI0OTAzNTE4M2E0YWYxNjc1MjE3MzQ2NTkxYjFjMDA1MjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FNctAHdUokIH_6fwa_FCAsbIq2PMSD991dXdab3rg6M">
> 
> Another thing regarding the search, I think we could add a feature to filter (kind of grep) the log lines to only display the search like t[his vs code plugin](https://marketplace.visualstudio.com/items?itemName=earshinov.filter-lines). We would provide some search templates like 'USER_DEBUG' AND 'EXCEPTION_THROWN'
> 
> Again, great work Samuel !

this arrow to navigate and all word that you find with the search like the ctrl f of the navigator @tprouvot

samuelkrissi committed:
Merge branch 'releaseCandidate' into feature/debugLogPage

samuelkrissi committed:
fix reviews and add ai analyse with promptTemplate

samuelkrissi commented:
HI @tprouvot 
I deactivated PRISME HIGHLIGHT where the file is bigger  than 1.5MB because is cause some issue of render when we open the preview 
only when we filter we can see the coloration

samuelkrissi committed:
fix unasable code and change max size to activate prismejs

samuelkrissi committed:
fix comment

samuelkrissi committed:
fix comment

samuelkrissi committed:
fix comment

samuelkrissi committed:
feat: Add customizable AI analysis instructions in debug log Agentforce modal

- Added edit mode toggle to switch between viewing and editing instructions
- Instructions are automatically saved to localStorage per org (sfHost)
- Added 'Edit' button to enter edit mode with brand styling when active
- Added 'Reset' button (only visible when customized) to restore defaults
- Added '✨ Customized' badge when instructions differ from defaults
- Edit mode shows textarea with 400px height and monospace font
- View mode displays instructions in read-only scrollable box
- All changes persist across sessions and are scoped to each Salesforce org
- Helpful tip text in edit mode guides users on customization
- Disabled controls during analysis to prevent concurrent modifications
- Extracted default instructions to getDefaultInstructions() method for reuse

tprouvot committed:
linter

tprouvot commented on CHANGES.md:
```diff
@@ -3,6 +3,7 @@
 ## Version 2.0
 
 - `Popup` Add "Copy Id" icon to User tab [feature 906](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/906) (contribution by [Arnav Gupta](https://github.com/ArnavGuptaaa)).
+- `Logs Viewer` Enhanced UI with pagination (10/15/25/50/100 logs per page), persistent page-size selector, "Logs (X of Y)" count with exact total, stable user filter populated from all logs org-wide, improved action label parsing (including Triggers with event names, VFRemote calls, and descriptive signatures), preview modal with search and highlight, grep-like log filtering with templates, AI/A ...
```
I think a short description like should be enough.
But this feature deserves a brand new .md file to detail and explain the capabilities of the page 😃 


```suggestion
- `Logs Viewer` Allow users to display, filter, delete, share and explain debug logs (contribution by [Samuel Krissi](https://github.com/samuelkrissi))
```

tprouvot committed:
Add setting in option page for log analysis prompt template name

tprouvot committed:
Fix meta extension for prompt templates and set AnalyzeDebugLog as active

tprouvot committed:
Rebrand AI with Agentforce and add option to hide buttons

tprouvot commented on addon/debug-log.js:
```diff
@@ -0,0 +1,1924 @@
+            h("use", {xlinkHref: "symbols.svg#einstein"})
+          )
+        ),
+        h("span", {}, "Agentforce Debug Log Analysis")
```
This is a great idea to use a prompt template here 👏 

There are minor design issues when the template does not exist and when we edit the prompt.
I think first modal should be hidden or greyed out.


<img width="1437" height="827" alt="Image" src="https://github.com/user-attachments/assets/d683beec-962b-4913-8962-44c6c02e465c" />

<img width="1342" height="735" alt="Image" src="https://github.com/user-attachments/assets/017132ed-77b3-4143-9cbc-ac6372d2c1c9" />

tprouvot commented:
Thanks for the updates @samuelkrissi, I love the prompt template usage 👍

samuelkrissi committed:
Update CHANGES.md

Co-authored-by: Thomas Prouvot <35368290+tprouvot@users.noreply.github.com>

samuelkrissi committed:
fix: Hide preview modal when Agentforce modal is open

- Added check to hide PreviewModal when showAgentforceModal is true
- Prevents UI confusion with overlapping modals
- Improves user experience by showing only one modal at a time
- Especially important when editing AI instructions

tprouvot commented on addon/debug-log.js:
```diff
@@ -0,0 +1,1927 @@
+              lineHeight: "1.7",
+              color: "#3e3e3c",
+              fontFamily: "system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, sans-serif",
+              padding: "0.75rem"
```
Could you keep default font for all the page and remove all fontFamily override ?

Could you use css instead of style property ?

tprouvot commented on addon/debug-log.js:
```diff
@@ -0,0 +1,1927 @@
+                padding: "1.25rem",
+                fontFamily: "'SF Pro Text', 'Segoe UI', system-ui, -apple-system, BlinkMacSystemFont, Roboto, sans-serif",
+                fontSize: "0.9375rem",
+                color: "#181818"
```
Could you keep default font for all the page and remove all fontFamily override ?

Could you use css instead of style property ?

tprouvot commented on addon/debug-log.js:
```diff
@@ -0,0 +1,1927 @@
+            className: "slds-textarea",
+            style: {
+              minHeight: "400px",
+              fontFamily: "system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, monospace",
```
Could you keep default font for all the page and remove all fontFamily override ?

Could you use css instead of style property ?

tprouvot commented on addon/debug-log.js:
```diff
@@ -0,0 +1,1927 @@
+    }
+  }
+
+  async loadMore() {
```
This method seems to be never called, could you confirm ?

tprouvot commented on addon/debug-log.js:
```diff
@@ -0,0 +1,1927 @@
+    }
+  }
+
+  async fetchTotalCount() {
```
This method seems to be never called, could you confirm ?

tprouvot committed:
Remove unnecessary emojis from debug log UI

tprouvot committed:
feat: Add sorting functionality to debug log table

tprouvot committed:
Harmonize debug log page size storage key

tprouvot commented:
We're almost there @samuelkrissi ! 
Merge is coming ... ❄️

samuelkrissi committed:
fix modal sizing
add button close like other modal on inspector
fix ui loading modal

samuelkrissi committed:
add log viewer md to explain

samuelkrissi committed:
fix inline css and remove unusual function

samuelkrissi committed:
Merge branch 'releaseCandidate' into feature/debugLogPage

samuelkrissi committed:
remove emojis

tprouvot committed:
Rename shortcut in manifest, implement keyboard shortcut from popup for logs viewer page and linter debug-logs

tprouvot committed:
Optimize populatePicklistFromAllLogs by using a single relationship query to fetch LogUserId and LogUser.Name, reducing the number of API calls

tprouvot committed:
Implement log body caching and searching, add toggle for fetching log bodies, and introduce new columns for operation and request details

tprouvot commented:
Thanks for the updates @skrissi 

Here are some updates I've made:

**API Optimization** (`ee2396a8`): Consolidated user name fetching from N+1 API calls to a single relationship query (`SELECT LogUserId, LogUser.Name FROM ApexLog`).

**Log Body Caching & Search** (`50075d2e`):
- Added toggle to enable/disable fetching log bodies (persisted in localStorage)
- Implemented client-side search within log bodies (filters displayed logs)
- Created `getLogBodyText()` method to centralize body retrieval with caching
- Preview/download/share actions now use cached bodies when available

**New Columns** (`50075d2e`):
- Added Operation and Request columns to the log table
- Moved Action column after Status column

**Code Quality** (`e6bff3da`): Keyboard shortcut implementation and linter fixes.

I have several remaining questions I wanted to discuss with you:
- Should we release this new page as '(beta)' to indicate users this is a new page that could have some bugs ?
- Since some logs can have a large size, should we limit the log body caching and searching for those large logs ? I think we could add a new option, predefined with 15 or 20mb that will allow users to lower this value in case of browser crash
- I was thinking of some improvements to reduce the number of api calls, for example adding some filters to the existing ones such as operation and LogLength 
- We could also add in a new modal some UI to enable logs for a particular user or update the log levels

Those are improvements that can be implemented later on and I'm ok with releasing the new page in the current state

Thanks for the job made ! 👏

samuelkrissi commented:
> Merci pour les mises à jour@skrissi
> 
> Voici quelques mises à jour que j'ai effectuées :
> 
> **Optimisation de l'API** ( `ee2396a8`) : Récupération consolidée du nom d'utilisateur à partir de N+1 appels d'API en une seule requête de relation ( `SELECT LogUserId, LogUser.Name FROM ApexLog`).
> 
> **Mise en cache et recherche du corps du journal** ( `50075d2e`):
> 
> * Ajout d'un bouton permettant d'activer/désactiver la récupération des corps des journaux (stockés dans localStorage).
> * Mise en place d'une recherche côté client dans les corps des journaux (filtres pour les journaux affichés)
> * Méthode créée `getLogBodyText()`pour centraliser la récupération du corps du corps avec mise en cache
> * Les actions de prévisualisation, de téléchargement et de partage utilisent désormais les corps mis en cache lorsqu'ils sont disponibles.
> 
> **Nouvelles colonnes** ( `50075d2e`):
> 
> * Ajout des colonnes Opération et Requête à la table de journalisation
> * Colonne Action déplacée après la colonne Statut
> 
> **Qualité du code** ( `e6bff3da`): Implémentation des raccourcis clavier et corrections du linter.
> 
> Il me reste plusieurs questions que je voulais aborder avec vous :
> 
> * Devrions-nous publier cette nouvelle page en tant que « (bêta) » pour indiquer aux utilisateurs qu'il s'agit d'une nouvelle page susceptible de contenir des bogues ?
> * Étant donné que certains journaux peuvent être volumineux, devrions-nous limiter la mise en cache et la recherche du corps de ces journaux ? Je pense qu'il serait judicieux d'ajouter une option prédéfinie à 15 ou 20 Mo, permettant aux utilisateurs de réduire cette valeur en cas de plantage du navigateur.
> * Je réfléchissais à des améliorations pour réduire le nombre d'appels API, par exemple en ajoutant des filtres aux filtres existants, tels que l'opération et la longueur du journal.
> * Nous pourrions également ajouter une nouvelle fenêtre modale ou une interface utilisateur permettant d'activer les journaux pour un utilisateur spécifique ou de modifier les niveaux de journalisation.
> 
> Ce sont des améliorations qui pourront être mises en œuvre ultérieurement et je suis d'accord pour publier la nouvelle page dans son état actuel.
> 
> Merci pour le travail accompli ! 👏

samuelkrissi commented:
thank you @tprouvot 
This is an excellent approach. Marking the page as "(Beta)" is the best way to manage user expectations while you refine advanced features (large file handling, API filters, logging) for a future update.

I'll update the button in popup.js to add the label "(Beta)".

tprouvot committed:
Unify md format with other pages

tprouvot commented on docs/logs-viewer.md:
```diff
@@ -0,0 +1,291 @@
+
+The Debug Logs Viewer allows you to view, filter, analyze, and manage Salesforce debug logs directly from the extension. With advanced features like Agentforce-powered analysis, grep-like filtering, and smart pagination, it transforms the debug log experience into a productive workflow.
+
+![Pagination](screenshots/logs-pagination.png)
```
Could you paste those screenshots in a comment from the PR and reference those images url in the .md file ?

tprouvot commented on docs/logs-viewer.md:
```diff
@@ -0,0 +1,291 @@
+
+Click the **Preview** button on any log to open the preview modal:
+
+![Log Preview](screenshots/logs-preview.png)
```
Could you paste those screenshots in a comment from the PR and reference those images url in the .md file ?

tprouvot commented on docs/logs-viewer.md:
```diff
@@ -0,0 +1,291 @@
+
+Transform your debugging experience with Agentforce-powered log analysis.
+
+![Agentforce Analysis](screenshots/logs-ai-analysis.png)
```
Could you paste those screenshots in a comment from the PR and reference those images url in the .md file ?

tprouvot commented on docs/logs-viewer.md:
```diff
@@ -0,0 +1,291 @@
+
+Customize the Agentforce analysis instructions to focus on what matters to you:
+
+![Custom Instructions](screenshots/logs-ai-custom-instructions.png)
```
Could you paste those screenshots in a comment from the PR and reference those images url in the .md file ?

samuelkrissi committed:
fix browser error in popup

samuelkrissi committed:
replace var by let

tprouvot commented:
Final screenshots:

Main page:
<img width="1437" height="827" alt="image" src="https://github.com/user-attachments/assets/2b916cb5-8ca8-4ebf-9d92-880b12d97d25" />

Log Preview:
<img width="1281" height="652" alt="Log Preview" src="https://github.com/user-attachments/assets/5714396c-1a45-4293-ae51-df5893294a7c" />

Log Analysis:
<img width="1262" height="656" alt="Log Analysis" src="https://github.com/user-attachments/assets/e5b3aa35-15bc-4611-91ed-ce7bbb550f70" />

Custom Analysis:
<img width="1272" height="395" alt="Customized Log Analysis" src="https://github.com/user-attachments/assets/114c96e8-43b3-41dc-99cc-b7032e046e76" />

tprouvot committed:
Replace screenshots
